Emulator/BF.File.Emulator/BfEmulator.cs

```diff
@@ -0,0 +1,217 @@
+                    _flowFormat = FlowFormatVersion.Version1;
+                    _library = LibraryLookup.GetLibrary("P4G");
+                    _encoding = AtlusEncoding.GetByName("P4");
+                    break;
+                case Game.P5R:
+                    _flowFormat = FlowFormatVersion.Version3BigEndian;
+                    _library = LibraryLookup.GetLibrary("P5R");
+                    _encoding = AtlusEncoding.GetByName("P5");
+                    break;
+            }
+        }
+
+        public bool TryCreateFile(IntPtr handle, string filepath, string route, out IEmulatedFile emulated)
+        {
+            // Check if we already made a custom BF for this file.
+            emulated = null!;
+            if (_pathToStream.TryGetValue(filepath, out var stream))
+            {
+                // Avoid recursion into same file.
+                if (stream == null)
+                    return false;
+
+                emulated = new EmulatedFile<MemoryManagerStream>(stream);
+                return true;
+            }
+
+            // Check extension.
+            if (!filepath.EndsWith(Constants.BfExtension, StringComparison.OrdinalIgnoreCase) || filepath.EndsWith(Constants.DumpExtension, StringComparison.OrdinalIgnoreCase))
+                return false;
+
+            if (!TryCreateEmulatedFile(handle, filepath, filepath, filepath, true, ref emulated!, out _))
+                return false;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Tries to create an emulated file from a given file handle.
+        /// </summary>
+        /// <param name="handle">Handle of the bf file where to use as a base.</param>
+        /// <param name="srcDataPath">Path of the file where the handle refers to.</param>
+        /// <param name="outputPath">Path where the emulated file is stored.</param>
+        /// <param name="route">The route of the emulated file, for builder to pick up.</param>
+        /// <param name="invokeOnStreamCreated">Invokes the <see cref="OnStreamCreated"/> event.</param>
+        /// <param name="emulated">The emulated file.</param>
+        /// <param name="stream">The created stream under the hood.</param>
+        /// <returns></returns>
+        public bool TryCreateEmulatedFile(IntPtr handle, string srcDataPath, string outputPath, string route, bool invokeOnStreamCreated, ref IEmulatedFile? emulated, out MemoryManagerStream? stream)
+        {
+            stream = null;
+
+            // Check if there's a known route for this file
+            // Put this before actual file check because I/O.
+            if (!_builderFactory.TryCreateFromPath(route, out var builder))
+                return false;
+
+            // Check file type.
+            if (!BfChecker.IsBfFileOrEmpty(handle, out var isEmpty))
+                return false;
+
+            // Make the BF file.
+            _pathToStream.TryRemove(outputPath, out _); // Avoid recursion into same file.
```

Comment:
This might be a bug (and it seems PAK emulator has it too), and should be replaced with a set to null.

This line is done such that the condition

```
if (_pathToStream.TryGetValue(filepath, out var stream))
{
    if (stream == null)
        return false;
```

above returns false; preventing recursion into a file for which emulation is being set up for.

Needs fixing for this and PAK emulator.

Reply:
Fixed in f2e57c1